orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -370,143 +371,176 @@ def test_input_requires_grad_backward_creates_input_grad(device):
+        run_step(pt_model1, pt_model2, x1, x2)
+        run_step(ort_model1, ort_model2, x1, x2)
 
-        assert x1.grad is not None and x2.grad is not None
-        for param in model1.parameters():
-            assert param.grad is not None
-            param.grad = None
-        for param in model2.parameters():
-            assert param.grad is not None
-            param.grad = None
+        _test_helpers.assert_gradients_match_and_reset_gradient(ort_model1, pt_model1)
+        _test_helpers.assert_gradients_match_and_reset_gradient(ort_model2, pt_model2)
 
 def test_multiple_ortmodules_common_backbone_training():
-    N, D_in, H, D_out = 32, 64, 500, 64
-    model = NeuralNetSinglePositionalArgument(D_in, H, D_out).to('cuda')
-    model1 = NeuralNetSinglePositionalArgument(D_in, H, D_out).to('cuda')
-    model2 = NeuralNetSinglePositionalArgument(D_in, H, D_out).to('cuda')
+    device = 'cuda'
+    N, D_in, H, D_out = 32, 64, 128, 64
+    pt_model0 = NeuralNetSinglePositionalArgument(D_in, H, D_out).to(device)
+    pt_model1 = NeuralNetSinglePositionalArgument(D_in, H, D_out).to(device)
+    pt_model2 = NeuralNetSinglePositionalArgument(D_in, H, D_out).to(device)
     # model is the common backbone shared by model1 and model2
-    model = ORTModule(model)   
-    model1 = ORTModule(model1)
-    model2 = ORTModule(model2)
+    ort_model0 = ORTModule(copy.deepcopy(pt_model0))
+    ort_model1 = ORTModule(copy.deepcopy(pt_model1))
+    ort_model2 = ORTModule(copy.deepcopy(pt_model2))
+
+    def run_step(backbone_layers, task_layers, x):
+        prediction = task_layers(backbone_layers(x))
+        loss = prediction.sum()
+        loss.backward()
 
     for step in range(10):
-        x1 = torch.randn(N, D_in, device='cuda', requires_grad=True)
-        x2 = torch.randn(N, D_in, device='cuda', requires_grad=True)
-        assert x1.grad is None and x2.grad is None
-
-        prediction1 = model1(model(x1))
-        s1 = prediction1.sum()
-        s1.backward()
-
-        prediction2 = model2(model(x2))
-        s2 = prediction2.sum()
-        s2.backward()
-
-        assert x1.grad is not None and x2.grad is not None
-        for param in model.parameters():
-            assert param.grad is not None
-            param.grad = None
-        for param in model1.parameters():
-            assert param.grad is not None
-            param.grad = None
-        for param in model2.parameters():
-            assert param.grad is not None
-            param.grad = None
+        # Run task 1
+        x1 = torch.randn(N, D_in, device=device)
+        run_step(pt_model0, pt_model1, x1)
+        run_step(ort_model0, ort_model1, x1)
+
+        _test_helpers.assert_gradients_match_and_reset_gradient(ort_model0, pt_model0, reset_gradient=False)
+        _test_helpers.assert_gradients_match_and_reset_gradient(ort_model1, pt_model1)
+
+        # Run task 2
+        x2 = torch.randn(N, D_in, device=device)
+        run_step(pt_model0, pt_model2, x1)
```

Comment:
Should be `x2` here, not `x1`?

Reply:
thanks for catching this. 
I will include this in my next PR